packages/main/src/plugin/cli-tool-registry.ts

```diff
@@ -82,17 +89,52 @@ export class CliToolRegistry {
     this.apiSender.send('cli-tool-remove', cliTool.id);
   }
 
-  getCliToolInfos(): CliToolInfo[] {
-    return Array.from(this.cliTools.values()).map(cliTool => {
-      return {
+  async getCliToolInfos(): Promise<CliToolInfo[]> {
+    const cliToolInfos: CliToolInfo[] = [];
+    for (const cliTool of this.cliTools.values()) {
+      const cliVersion = await cliTool.getVersion();
+      cliToolInfos.push({
         id: cliTool.id,
         name: cliTool.name,
         displayName: cliTool.displayName,
         description: cliTool.markdownDescription,
         state: cliTool.state,
         images: cliTool.images,
         extensionInfo: cliTool.extensionInfo,
-      };
-    });
+        version: cliVersion,
+      });
+    }
+    return cliToolInfos;
   }
 }
+
+// This function will take in the exec, name and storage path of the binary.
+// it will also take in the "command" that achieves getting the version of the binary as
+// each tool has a different way of getting it (for example, -v, --version, version, --help, etc.)
+// NOTE: We only care what's in the ~/.local/share/containers/podman-desktop/extensions-storage/ folder
+// not the binary that is installed on the system (since we have the ability to install system-wide)
+async function getCliVersion(exec: Exec, name: string, storagePath: string, helpCommand: string): Promise<string> {
```

Comment:
IMO it should be declared in CliTool interface and implemented in CliToolImpl. The reason is it would need access to CliToolImpl internal properties to change state of CliToolImpl instance. It has currently only one state `registered`. It means CliTool added podman desktop and nothing was done with it. As we introduce detection there are going to be additional states 'detection is in progress`, 'not detected', 'detected'. Those can be calculated based on values of internal properties (if version property is present then state is detected and etc) or set explicitly by detection function as it goes through different detection stages.